modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/KeyIdResolverModelParser.java

```diff
@@ -36,4 +37,18 @@ public interface KeyIdResolverModelParser {
    */
   TypeKeysResolver getKeyResolver();
 
+  /**
+   * @return the class of the resolver
+   */
+  Class<?> keyIdResolverDeclarationClass();
```

Comment:
This parser will need to be implemented by any sdk that implements custom metadata. Returning a java Class seems to be an issue. I think it is more appropiate to either ask for an instance of a Resolver, or a Supplier of Resolvers.